.github/workflows/release.yml

```diff
@@ -110,4 +110,27 @@ jobs:
           collie info
       - name: test
         run: |
-          ./test/e2e.sh
\ No newline at end of file
+          ./test/e2e.sh
+  e2e-macos:
+    runs-on: macos-latest
+    needs: artifacts
+    steps:
+      - uses: actions/checkout@v4
+      - name: install dependencies 
+        # note: we forego default.nix and to be as close to a realistic user experience as possible
+        # all the other dependencies are already present on the runner images, so we use those
+        # and just install the missing ones via nix
+        uses: cachix/install-nix-action@v22
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          extra_nix_config: |
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      - uses: workflow/nix-shell-action@v3.3.0
```

Comment:
I assume the rrbutani action doesn't work on macos?

other thatn that I'd jus ttry to keep parity between the steps used on linux and mac (break down of shell script into gh build steps for quick visibility when opening a failed build. Though that's a matter of taste. But consistency is king.